File: 9-palindrome-number/9-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if(x < 0) return false;
        
        int rev = 0, rem, orig = x;
        
        while(x != 0) {
            rem = x % 10;
            rev = (rev * 10) + rem;
            x /= 10;
        }
        
        return rev == orig;
    }
}

File: 9-palindrome-number/9-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        
        if(x < 0)
            return 0;
        if(x < 9)
            return 1;
        int n = x;
        long long rx = 0;
        while (n) {
            rx = (n%10) + (rx*10);
            n /= 10;
        }
        cout<<rx;
        if((int)rx == x)
            return 1;
        
        return 0;
    }
};